packages/babel-plugin/src/utils/preserve-flow-comment.tsx

```diff
@@ -0,0 +1,20 @@
+import type { NodePath } from '@babel/traverse';
+import type * as t from '@babel/types';
+
+/**
+ * Ensure that any flow comment is preserved at the top of the processed file before additional imports are
+ * dynamically inserted during the build.
+ *
+ * @param path
+ */
+export const preserveFlowComment = (path: NodePath<t.Program>): void => {
+  const leadingComment =
+    path.node.body.length > 0 &&
+    path.node.body[0].leadingComments &&
+    path.node.body[0].leadingComments.length > 0 &&
+    path.node.body[0].leadingComments[0];
```

Comment:
I think we can do this in 1 statement with optional chaining. Something like:
`path.node.body?.[0].leadingComments?.[0]`

Reply:
Flow is not guaranteed to be the top comment either, it's in any leading comment

Reply:
@at-nathan Yes, we can. When I tried it, I was getting some errors, maybe I had a typo

@MonicaOlejniczak fair point, I had the assumption that we always place flow at the top